[DOCS] DSB-484 Improve Quickstart diagram links

rachhouse opened the pull request:
This PR addresses feedback coming from one of our Champions that the current Quickstart diagram links are confusing. See this ticket for more details: https://greatexpectations.atlassian.net/browse/DSB-484

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

rachhouse committed:
update diagram to direct to on-page steps, add related documentation links

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8951&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/8951/27a93c6b/75d7be33002fda234db0430f0d0df58f74d85aa5.svg)](https://app.codesee.io/r/reviews?pr=8951&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

rachhouse commented:
@Rachel-Reverie @kwcanuck This PR addresses feedback coming from one of our Champions that the current Quickstart diagram links are very confusing.

After considering [Cesar's feedback on the current Quickstart diagram links](https://greatexpectationstalk.slack.com/archives/CUW3WD6PK/p1699531411972709), I've made the following changes:
* The diagram workflow step boxes link to the related step on the OSS Quickstart page itself rather than the separate, detailed docs pages.
* The *Related documentation* section now contains links to the more detailed docs pages for the key Quickstart steps.

Here's a direct link to the Quickstart preview:
https://deploy-preview-8951.docs.greatexpectations.io/docs/tutorials/quickstart/

I think these are merited improvements because:
* Cesar has strong point that new users may be confused if the diagram directs them to another page. I had not considered the potential confusion produced by a redirect.
* The original intent behind the Quickstart diagram was to:
  * Express a basic GX workflow and build user intuition on how the GX components fit together.
  * Match the current docs TOC and help users understand the layout of the TOC in relation to GX workflow steps.

  Based on our conversation earlier this week, we have a goal of expressing documentation in *concept-task-reference* format and providing conceptual, longer-form educational content in our growing *Concepts*/*Learn* section in the docs. I believe these changes clarify the content of the Quickstart to better adhere to *concept-task-reference*, and I think that our intended growth of *Concepts*/*Learn* will enable us to include the conceptual workflow information with which I originally tried to imbue the Quickstart. I think by trying to educate within the Quickstart, I made things more confusing.

Let me know if you have any concerns. I would also welcome your opinions on whether we need to update the Quickstart section headers to more closely match the diagram workflow step text.

kwcanuck commented on docs/docusaurus/docs/tutorials/quickstart/quickstart.md:
```diff
@@ -28,34 +28,34 @@ Windows support for the open source Python version of GX is currently unavailabl
 ## Data validation workflow
 
-The following diagram illustrates the end-to-end GX data validation workflow implemented by the quickstart code. Click a workflow step to view related documentation.
+The following diagram illustrates the end-to-end GX data validation workflow implemented by the quickstart code. Click a workflow step to view related code.
```
```suggestion
The following diagram illustrates the end-to-end GX data validation workflow that you'll implement with this quickstart. Click a workflow step to view the related content.
```

kwcanuck commented on docs/docusaurus/docs/tutorials/quickstart/quickstart.md:
```diff
@@ -124,7 +124,10 @@ click Validate_B "https://docs.greatexpectations.io/docs/terms/data_docs"
 ## Related documentation
 
-If you're ready to continue your Great Expectations journey, the following topics can help you implement a tailored solution for your specific environment and business requirements:
+If you're ready to continue your GX journey, the following topics can help you implement a tailored solution for your specific environment and business requirements:
```
```suggestion
If you're ready to continue your GX journey, the following topics can help you implement a solution for your specific environment and business requirements:
```

rachhouse committed:
Apply suggestions from code review

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>
